decompiler/structures/ast/reachability_graph.py

```diff
@@ -258,7 +258,7 @@ def has_path(self, source: AbstractSyntaxTreeNode, sink: AbstractSyntaxTreeNode,
         assert source in self._case_node_reachability_graph.nodes, f"Source {source} not in the graph"
         assert sink in self._case_node_reachability_graph.nodes, f"Sink {sink} not in the graph."
         removed_edge = False
-        if no_edge and sink in self.reachable_cases_of(source):
+        if no_edge and sink in set(self.reachable_cases_of(source)):
```

Comment:
Yes, this is because hash checks whether these are different nodes in the same graph, and eq checks whether they are of the same type and have the same attributes.